doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst

```diff
@@ -652,6 +652,87 @@ The following changes are recommended for your application to work optimally aft
+  * The :c:struct:`bt_mesh_sensor_value`.format needs to be filled by the application for variables passed to the Sensor API.
+  * There are several different types of :c:struct:`bt_mesh_sensor_format` described in the :file:`include/bluetooth/mesh/sensor_types.h`.
+  * When filling in sensor values for a channel the format can be found through ``sensor->type.channels[i].format`` defined for the given :c:struct:`bt_mesh_sensor` ``sensor``.
+  * The :c:struct:`bt_mesh_sensor_format` with a valid `format` can be converted to and from integer, float and :c:struct:`sensor_value` through `bt_mesh_sensor_value_to/from*` functions.
+  * Where the applications previously just added values directly to :c:struct:`sensor_value`.val1 and .val2, now the correct way is to use `bt_mesh_sensor_value_to/from*` functions to either set/extract values.
+
+  Example of changes need to be done for sensor using sensor_values, from e.g. drivers/sensor.h
+
+    ..  code-block:: diff
+
+        static int chip_temp_get(struct bt_mesh_sensor_srv *srv,
+                                 struct bt_mesh_sensor *sensor,
+                                 struct bt_mesh_msg_ctx *ctx,
+        -                         struct sensor_value *rsp)
+        +                         struct bt_mesh_sensor_value *rsp)
+        {
+        +        struct sensor_value channel_val;
+                int err;
+
+                sensor_sample_fetch(dev);
+
+        -        err = sensor_channel_get(dev, SENSOR_DATA_TYPE, rsp);
+        +        err = sensor_channel_get(dev, SENSOR_DATA_TYPE, &channel_val);
+                if (err) {
+                        printk("Error getting temperature sensor data (%d)\n", err);
+                }
+        +        err = bt_mesh_sensor_value_from_sensor_value(
+        +                sensor->type->channels[0].format, &channel_val, rsp);
+        +        if (err) {
+        +                printk("Error encoding temperature sensor data (%d)\n", err);
+        +        }
+
+                return err;
+        }
+
+  Example of changes need to be done for sensor using float values.
+
+    ..  code-block:: diff
+
+        static int amb_light_level_get(struct bt_mesh_sensor_srv *srv,
+                                       struct bt_mesh_sensor *sensor,
+                                       struct bt_mesh_msg_ctx *ctx,
+        -                               struct sensor_value *rsp)
+        +                               struct bt_mesh_sensor_value *rsp)
+        {
+                int err;
+
+                /* Report ambient light as dummy value, and changing it by pressing
+                 * a button. The logic and hardware for measuring the actual ambient
+                 * light usage of the device should be implemented here.
+                 */
+        -        double reported_value = amb_light_level_gain * dummy_ambient_light_value;
+        +        float reported_value = amb_light_level_gain * dummy_ambient_light_value;
```

Comment:
I would rather have it like the way it is. Because that matches what is done in samples. That way people can actual match this up against what was done for our sample to see the difference. It then does becomes more real.